text/0097-log-data-model.md

```diff
@@ -0,0 +1,971 @@
+  format. An example is Apache log file.
+
+- First-party Applications. These are applications that we develop and we have
+  some control over how the logs and events are generated and what information
+  we include in the logs. We can likely modify the source code of the
+  application if needed.
+
+### Field Kinds
+
+This Data Model defines a logical model for a log record (irrespective of the
+physical format and encoding of the record). Each record contains 2 kinds of
+fields:
+
+- Named top-level fields of specific type and meaning.
+
+- Fields stored in the key/value pair lists, which can contain arbitrary values
+  of different types. The keys and values for well-known fields follow semantic
+  conventions for key names and possible values that allow all parties that work
+  with the field to have the same interpretation of the data. See references to
+  semantic conventions for `Resource` and `Attributes` fields and examples in
+  [Appendix A](#appendix-a-example-mappings).
+
+The reasons for having these 2 kinds of fields are:
+
+- Ability to efficiently represent named top-level fields, which are almost
+  always present (e.g. when using encodings like Protocol Buffers where fields
+  are enumerated but not named on the wire).
+  
+- Ability to enforce types of named fields, which is very useful for compiled
+  languages with type checks.
+
+- Flexibility to represent less frequent data via key/value pair lists. This
+  includes well-known data that has standardized semantics as well as arbitrary
+  custom data that the application may want to include in the logs.
+
+When designing this data model I followed the following reasoning to make a
+decision about when to use use a top-level named field:
+
+- The field needs to be either mandatory for all records or be frequently
+  present in well-known log and event formats (such as `Timestamp`) or is
+  expected to be often present in log records in upcoming logging systems (such
+  as `TraceId`).
+
+- The field’s semantics must be the same for all known log and event formats and
+  can be mapped directly and unambiguously to this data model.
+  
+Both of the above conditions were required to give the field a place in the
+top-level structure of the record.
+
+## Log and Event Record Definition
+
+Note: below we use type `any`, which can be a scalar value (number, string or
+boolean), or an array or map of values. Arbitrary deep nesting of values for
+arrays and maps is allowed (essentially allow to represent an equivalent of a
+JSON object).
+
+[Appendix A](#appendix-a-example-mappings) contains many examples that show how
+existing log formats map to the fields defined below. If there are questions
+about the meaning of the field reviewing the examples may be helpful.
+
+Here is the list of fields in a log record:
+
+Field Name     |
+---------------|
+Timestamp      |
+TraceId        |
+SpanId         |
+TraceFlags     |
```

Comment:
Do we want `TraceFlags` here?

Reply:
I had people requesting this because it will be useful to know if the trace was sampled (which is indicated in the TraceFlags). This for example will allow the UIs to know that the trace corresponding to this log actually exists too and can be linked to.

Reply:
FWIW, the New Relic Trace Context implementations ignore the inbound TraceFlags and we make our own sampling decision, captured in our tracestate. The outbound TraceFlags, however, do reflect the New Relic sampling decision. Because of this, including TraceFlags wouldn't be a good signal of the trace being sampled. 

However, I agree that TraceFlags should be included for futureproofing. I would hate to see w3c add a useful flag and we have to change our specs to include them.

Reply:
I don't want to derail this thread but that is a concerning thing to learn about New Relic implementations. Is this simply because New Relic hasn't fully integrated with otel because otel is still in its early stages? As opposed to being the plan to continue as a separate implementation that ignores inbound trace flags.

Reply:
Resolving, no further action needed.